src/main/java/bio/terra/service/snapshotbuilder/SnapshotRequestDao.java

```diff
@@ -0,0 +1,186 @@
+package bio.terra.service.snapshotbuilder;
+
+import bio.terra.common.DaoKeyHolder;
+import bio.terra.common.exception.BadRequestException;
+import bio.terra.common.exception.NotFoundException;
+import bio.terra.model.SnapshotAccessRequest;
+import bio.terra.model.SnapshotAccessRequestResponse;
+import bio.terra.model.SnapshotBuilderRequest;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+@Repository
+public class SnapshotRequestDao {
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+  private static final SnapshotAccessRequestResponseMapper responseMapper =
+      new SnapshotAccessRequestResponseMapper();
+  private static final String requestIdField = "id";
+  private static final String datasetIdField = "dataset_id";
+  private static final String requestNameField = "request_name";
+  private static final String requestResearchPurposeField = "request_research_purpose";
+  private static final String snapshotBuilderRequestField = "snapshot_builder_request";
+  private static final String userEmailField = "user_email";
+  private static final String createDateField = "create_date";
+  private static final String updateDateField = "update_date";
+  private static final String statusField = "status";
+
+  @Autowired
+  public SnapshotRequestDao(NamedParameterJdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  private static class SnapshotAccessRequestResponseMapper
+      implements RowMapper<SnapshotAccessRequestResponse> {
+    public SnapshotAccessRequestResponse mapRow(ResultSet rs, int rowNum) throws SQLException {
```

Comment:
You don't need to create an class for this mapper, you can use an anonymous one and create the mapper directly.
```
  private static final RowMapper<SnapshotAccessRequestResponse> responseMapper =
      (rs, rowNum) ->
          new SnapshotAccessRequestResponse()
              .id(rs.getObject(requestIdField, UUID.class))
              .datasetId(rs.getObject(datasetIdField, UUID.class))
              .requestName(rs.getString(requestNameField))
              .requestResearchPurpose(rs.getString(requestResearchPurposeField))
              .request(mapRequestFromJson(rs.getString(snapshotBuilderRequestField)))
              .createDate(getInstantString(rs, createDateField))
              .updateDate(getInstantString(rs, updateDateField))
              .userEmail(rs.getString(userEmailField))
              .status(SnapshotAccessRequestResponse.StatusEnum.valueOf(rs.getString(statusField)));
```
This works although it does require that you make `getInstantString()` `static`.

Reply:
Yes, or just `MAPPER`, typically you only have one per DAO.